### ISSUE TITLE
logind leaves tty in a bad state (no keyboard input possible) after suspend/resume

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#15880]
To reproduce the problem, have a laptop at hand. Boot it, log in as normal user, switch to say tty3 and login on the console. Close the lid -> system suspend. Open the lid -> system resumes. At this point, no keyboard input is possible anymore on tty3.

I could reproduce this behaviour on Debian sid, Fedora 32 and Ubuntu 20.04.

**Comment by mbiebl** [comment]
If I simply run `sudo echo mem > /sys/power/state` on tty3, the system resumes properly. Which leads me to believe, there is same bad logind interaction going on.

**Comment by mbiebl** [comment]
I don't remember seeing this issue with older systemd versions, so tentatively marking that as a regression.

**Comment by poettering** [comment]
Is this still an issue?

is this a text login? is there anything graphical involved? what makes you think logind is at fault? anything in the logs?

**Comment by mbiebl** [comment]
It's still reproducible with v247 with the steps outlined in the initial message. After resume, no input on tty3 is possible.

**Comment by mbiebl** [comment]
@poettering Just curious, is this not reproducible for you with the steps outlined above? I explicitly tested this on Fedora as well.

**Comment by poettering** [comment]
didn't try. 

but still: is this a text login? anything graphical?

what makes you think logind is at fault, and not kernel?

**Comment by mbiebl** [comment]
> but still: is this a text login? anything graphical?
> 
> what makes you think logind is at fault, and not kernel?

This is all answered in my previous posts

**Comment by mbiebl** [comment]
> didn't try.

Why not?

**Comment by poettering** [comment]
> This is all answered in my previous posts

Where? when i search through this page i only find my own comments with the keywords "text" and "graphical", and no answer...

> > didn't try.
> 
> Why not?

I suspend all the time of course, never had keyboard problems. I switch forth and back from tty3 to my gnome session, all is good.

So let me ask again: is this on the console with no graphical session whatsoever around? you booted up without gdm? 

not that kernel suspend implementations do some tty switching themselves, without logind being involved.

anything in the logs?

**Comment by mbiebl** [comment]
Please try the steps I outlined in post 1

**Comment by mbiebl** [comment]
> To reproduce the problem, have a laptop at hand. Boot it, log in as normal user,

log in via gdm (i.e. graphical) into GNOME as userA

> switch to say tty3 and login on the console. 

log in on the console (i.e. text) as userA

> Close the lid -> system suspend. Open the lid -> system resumes. At this point, no keyboard input is possible anymore on tty3.

No keyboard input possible on "text" login

**Comment by mbiebl** [comment]
> So let me ask again: is this on the console with no graphical session whatsoever around? you booted up without gdm?

No. I booted with gdm enabled, logged in, switched to tty3, logged in, suspended, resumed -> no input possible anymore.
I literally said that in my first post.

> not that kernel suspend implementations do some tty switching themselves, without logind being involved.

https://github.com/systemd/systemd/issues/15880#issuecomment-632295688

**Comment by poettering** [comment]
> > didn't try.
> 
> Why not?

didn't want to interrupt my work with a hung machine... I am mostly trying to figure out if this looks like another incident of #8522 in a different situation, but for that i need to understand what's going on.
